### PR TITLE
Add vector.dot and vector.cross

### PR DESCRIPTION
**Pull request by HybridDog**
Mostly copied from MarkuBu's code

Add compact, short information about your PR for easier understanding:

- Goal of the PR: Add frequently used dot and cross functions
- How does the PR work? The functions are added in the respective file and documented in lua_api.txt.
- Does it resolve any reported issue? Yes, it solves #8667.
- If not a bug fix, why is this PR needed? What usecases does it solve? Many mods use the dot product, so often they need to define it themselves.

## To do

This PR is Ready for Review.

- [x] List
- [x] Things
- [x] To do

## How to test

Please test the functions.

### CODE REVIEW

**Comment by MarkuBu** [comment]
I'm glad to see that my work won't be lost.

**Comment by paramat** [comment]
Seems useful.

**Comment by stujones11** [comment]
Definitely a worthwhile inclusion :+1: 

It might also be nice if you could reduce some code duplication by re-using these method here:
https://github.com/minetest/minetest/blob/3d617aa47275282bc8b5f79bb92627bc8a06653c/builtin/common/vector.lua#L74-L77

File: builtin/common/vector.lua [hunk 1]
```diff
@@ -71,14 +71,24 @@ function vector.direction(pos1, pos2)
...
-	local cpz = a.x * b.y - a.y * b.x
-	local crossplen = math.sqrt(cpx ^ 2 + cpy ^ 2 + cpz ^ 2)
+	local dotp = vector.dot(a, b)
+	local cp = vector.cross(a, b)
+	local crossplen = vector.length(cp)
 	return math.atan2(crossplen, dotp)
 end
 
+function vector.dot(a, b)
+	return a.x * b.x + a.y * b.y + a.z * b.z
```
Comment (DS-Minetest) [inline builtin/common/vector.lua]: Should 2D vectors be supported?
```suggestion
	return a.x * b.x + a.y * b.y + (a.z or 0) * (b.z or 0)
```

Comment (SmallJoker) [reply builtin/common/vector.lua]: Minetest vectors are fixed to 3D. Set the z coordinate manually to 0 if you'd like to use the functions for 2D vectors.

**Pull request merged**